compiler/rustc_const_eval/src/const_eval/fn_queries.rs

```diff
@@ -15,20 +15,14 @@ fn parent_impl_constness(tcx: TyCtxt<'_>, def_id: LocalDefId) -> hir::Constness
     }
 }
 
-/// Checks whether an item is considered to be `const`. If it is a constructor, it is const.
-/// If it is an assoc method or function,
-/// return if it has a `const` modifier. If it is an intrinsic, report whether said intrinsic
-/// has a `rustc_const_{un,}stable` attribute. Otherwise, panic.
+/// Checks whether a function-like thing is considered to be `const`.
 fn constness(tcx: TyCtxt<'_>, def_id: LocalDefId) -> hir::Constness {
     let node = tcx.hir_node_by_def_id(def_id);
 
     match node {
```

Comment:
Since #134122 this fn/query is only meant to work on function-likes. However on master it works on impl associated constants[^1] and all foreign items (should be: foreign functions only).

This rectifies it.

[^1]: Lol, not even trait+impl associated constants smh!